### PR TITLE
Avoid overflow in settings

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

See https://github.com/ClickHouse/ClickHouse/pull/60087
https://s3.amazonaws.com/clickhouse-test-reports/60087/d10b18afeb121b47b7ccef384550e9eb765391e2/ast_fuzzer__asan_.html

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit b5efbe62e7dc25f6049571348ccf18d34ba5a191 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60089/b5efbe62e7dc25f6049571348ccf18d34ba5a191/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/60089/b5efbe62e7dc25f6049571348ccf18d34ba5a191/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: tests/queries/0_stateless/01557_field_infinite_convert_to_number.sql [hunk 1]
```diff
@@ -1 +1 @@
-SET max_threads = nan; -- { serverError 70 }
+SET max_threads = nan; -- { serverError 69 }
```
Comment (yakov-olkhovskiy) [inline tests/queries/0_stateless/01557_field_infinite_convert_to_number.sql:1]: 70 is CANNOT_CONVERT_TYPE, 69 is ARGUMENT_OUT_OF_BOUND - should it be 70 still? also, we can put string here:
```suggestion
SET max_threads = nan; -- { serverError CANNOT_CONVERT_TYPE }
```

File: tests/queries/0_stateless/02992_settings_overflow.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+SET max_threads = -1; -- { serverError ARGUMENT_OUT_OF_BOUND }
```
Comment (yakov-olkhovskiy) [inline tests/queries/0_stateless/02992_settings_overflow.sql:1]: the same - should be CANNOT_CONVERT_TYPE

**Comment by clickgapai** [comment]
Hi @alexey-milovidov — while reviewing this PR I found the following:

- **Coverage gap:** Float64 finite-out-of-range branch in `fieldToNumber` (`SettingsFields.cpp:119`) is untested for setting assignment — https://github.com/ClickHouse/ClickHouse/pull/103863

Could you review the test PR and add the `can be tested` label if it looks good?
Happy to discuss — close anything that's wrong or already addressed.

**Pull request merged**